src/System.Drawing.Common/tests/mono/System.Drawing.Imaging/IconCodecTests.cs

```diff
@@ -0,0 +1,2005 @@
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Security.Permissions;
+using System.Text;
+using Xunit;
+
+namespace MonoTests.System.Drawing.Imaging
+{
+
+    public class IconCodecTest
+    {
+
+        /* Get suffix to add to the filename */
+        internal string GetOutSufix()
+        {
+            string s;
+
+            int p = (int)Environment.OSVersion.Platform;
+            if ((p == 4) || (p == 128) || (p == 6))
+                s = "-unix";
+            else
+                s = "-windows";
+
+            if (Type.GetType("Mono.Runtime", false) == null)
+                s += "-msnet";
+            else
+                s += "-mono";
+
+            return s;
+        }
+
+        [ConditionalFact(Helpers.RecentGdiplusIsAvailable)]
+        public void Image16()
+        {
+            string sInFile = Helpers.GetTestBitmapPath("16x16_one_entry_4bit.ico");
+            using (Image image = Image.FromFile(sInFile))
+            {
+                Assert.True(image.RawFormat.Equals(ImageFormat.Icon));
+                // note that image is "promoted" to 32bits
+                Assert.Equal(PixelFormat.Format32bppArgb, image.PixelFormat);
+                Assert.Equal(73746, image.Flags);
+
+                // The values are inconsistent across Windows & Unix: GDI+ returns 0, libgdiplus returns 16.
+                // Assert.Equal(16, image.Palette.Entries.Length);
+
+                using (Bitmap bmp = new Bitmap(image))
+                {
+                    Assert.True(bmp.RawFormat.Equals(ImageFormat.MemoryBmp));
+                    Assert.Equal(PixelFormat.Format32bppArgb, bmp.PixelFormat);
+                    Assert.Equal(2, bmp.Flags);
+                    Assert.Equal(0, bmp.Palette.Entries.Length);
+                }
+            }
+        }
+
+        // simley.ico has 48x48, 32x32 and 16x16 images (in that order)
+        [ConditionalFact(Helpers.RecentGdiplusIsAvailable)]
+        public void Bitmap16Features()
+        {
+            string sInFile = Helpers.GetTestBitmapPath("48x48_multiple_entries_4bit.ico");
+            using (Bitmap bmp = new Bitmap(sInFile))
+            {
+                GraphicsUnit unit = GraphicsUnit.World;
+                RectangleF rect = bmp.GetBounds(ref unit);
+
+                Assert.True(bmp.RawFormat.Equals(ImageFormat.Icon));
+                // note that image is "promoted" to 32bits
+                Assert.Equal(PixelFormat.Format32bppArgb, bmp.PixelFormat);
+                Assert.Equal(73746, bmp.Flags);
+
+                // These values are inconsistent accross Windows & Unix: 0 on Windows, 16 on Unix
```

Comment:
We shouldn't leave commented-out code like this in the tests. If we are able to, we can move it into a Windows-specific test case, and disable it on Unix. If we don't do that, we should just remove the dead code.

Reply:
@mellinoe I created two test cases, one for the Windows behavior and one for the Unix behavior. I did it because the Unix behavior is in a sense more complete: Unix returns palette data whereas Windows does not.

It also ensures getting the Palette will keep on working on Unix (as opposed to disabling the test altogether so that the code will never be exercised).